### PR TITLE
Add clients for harbor

### PR DESCRIPTION
**Pull request by lubronzhan**
Including 3 clients:
v2: newest 2.4 client
legacy: legacy client
assist: contains client for chart heathcheck or version

`pkg/harbor/client.go` is a client wrapper for creating these 3 harbor clients

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+HARBOR_ASSIST_SPEC=api/swagger.yaml
+HARBOR_CLIENT_ASSIST_DIR=pkg/sdk/assist
+
+HARBOR_2.0_SPEC=api/v2.0/swagger.yaml
+HARBOR_CLIENT_2.0_DIR=pkg/sdk/v2.0
+
+HARBOR_2.0_LEGACY_SPEC=api/v2.0/legacy_swagger.yaml
+HARBOR_CLIENT_2.0_LEGACY_DIR=pkg/sdk/v2.0/legacy
+
+## TODO: pull the newest swagger file based on version from harbor repository
```
Comment (steven-zou) [inline Makefile:13]: Raise another PR to complete it?

File: pkg/harbor/client.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	"net/url"
+	"time"
+
+	httptransport "github.com/go-openapi/runtime/client"
+	assistclient "github.com/goharbor/go-client/pkg/sdk/assist/client"
+	v2client "github.com/goharbor/go-client/pkg/sdk/v2.0/client"
+	legacyclient "github.com/goharbor/go-client/pkg/sdk/v2.0/legacy/client"
+)
+
+// insecureTransport provides a insecure RoundTripper and disable the HTTP2 try.
```
Comment (steven-zou) [inline pkg/harbor/client.go:16]: I think it should be something like this:

```go
// Use one of the config to alias the config here.
type Config v2client.Config

func ( c *Conifg) ToAssistConfig()*assistclient.Config{
  if c == nil {
   return nil
 }
  // TODO: do value copy here
}

func ( c *Conifg) ToV2Config()*v2client.Config{
  // TODO: do value copy here
}

func ( c *Conifg) ToLegacyConfig()*legacyclient.Config{
  // TODO: do value copy here
}

// Define the client for using.
type Client struct {
  aClient *assistclient.HarborAPI
  v2Client *v2client.HarborAPI
  lClient *legacyclient.HarborAPI
}

// Client with config.
type ClientWithConfig struct {
  Client
  config *Config
}

// Return client set references.
func ClientSet() *Client{
  return &Client{}
}

func (c *Client) WithConfig(config *Config) *ClientWithConfig
  return &ClientWithConfig{
    config: config,
  }
}

func (cc *ClientWithConfig) AssisClient() *assistclient.HarborAPI {
  if cc.aClient == nil {
   cc.aClient = assistclient.New(cc.config.ToAssistConfig())
  }
 
  return cc.aClient
}

func (cc *ClientWithConfig) V2Client() *v2client.HarborAPI {
  if cc.v2Client == nil {
   cc.v2Client = assistclient.New(cc.config.ToV2Config())
  }
 
  return cc.v2Client
}

func (cc *ClientWithConfig) LegacyClient() *legacyclient.HarborAPI {
  if cc.lClient == nil {
   cc.lClient = assistclient.New(cc.config.ToLegacyConfig())
  }
 
  return cc.lClient
}
```

**Review by steven-zou — COMMENTED** [review]
Check my comments.

File: Makefile (hunk 1 above)
Comment (lubronzhan) [reply Makefile:13]: yes

**Review by bitsf — APPROVED** [review]

**Review by steven-zou — APPROVED** [review]
LGTM

**Pull request merged**